The following is a list of 10 students ages:
ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
• Sort the list and find the min and max age
• Add the min age and the max age again to the list
• Find the median age (one middle item or two middle items divided by two)
• Find the average age (sum of all items divided by their number)
• Find the range of the ages (max minus min)

In [17]:
# Given list of ages
ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]

# Sort the list and find the min and max age
ages.sort()
min_age = ages[0]
max_age = ages[-1]
print("Sorted ages:", ages)
print("Minimum age:", min_age)
print("Maximum age:", max_age)

# Add the min age and the max age again to the list
ages.extend([min_age, max_age])
print("ages after adding minimum and maximum again:", ages)


# Find the median age
ages.sort()
n = len(ages)
if n % 2 == 0:
    median_age = (ages[n // 2 - 1] + ages[n // 2]) / 2
else:
    median_age = ages[n // 2]
print("Median age:", median_age)

# Find the average age
average_age = sum(ages) / n
print("Average age:", average_age)

# Find the range of the ages
age_range = max_age - min_age
print("ranges of age:", age_range)


Sorted ages: [19, 19, 20, 22, 24, 24, 24, 25, 25, 26]
Minimum age: 19
Maximum age: 26
ages after adding minimum and maximum again: [19, 19, 20, 22, 24, 24, 24, 25, 25, 26, 19, 26]
Median age: 24.0
Average age: 22.75
ranges of age: 7


Create an empty dictionary called dog
• Add name, color, breed, legs, age to the dog dictionary
• Create a student dictionary and add first_name, last_name, gender, age, marital status,
skills, country, city and address as keys for the dictionary
• Get the length of the student dictionary
• Get the value of skills and check the data type, it should be a list
• Modify the skills values by adding one or two skills
• Get the dictionary keys as a list
• Get the dictionary values as a list

In [19]:
# Create an empty dictionary called dog
dog = {}

# Add name, color, breed, legs, age to the dog dictionary
dog['name'] = 'preethi'
dog['color'] = 'white'
dog['breed'] = 'Labrador'
dog['legs'] = 4
dog['age'] = 3
print("dog dictionary:", dog)

# Create a student dictionary and add various keys
student = {
    'first_name': 'sai',
    'last_name': 'yeminedi',
    'gender': 'Male',
    'age': 24,
    'marital_status': 'Single',
    'skills': ['Python', 'Java'],
    'country': 'USA',
    'city': 'kansas',
    'address': '525 E armour blvd'
}
print("student dictionary:", student)

# Get the length of the student dictionary
student_length = len(student)
print("Length of student dictionary:", student_length)

# Get the value of skills and check the data type, it should be a list
skills = student['skills']
print("Skills:", skills)
print("Data type of skills:", type(skills))

# Modify the skills values by adding one or two skills
student['skills'].append('JavaScript')
student['skills'].append('HTML')
print("updated skills:", student['skills'])

# Get the dictionary keys as a list
student_keys = list(student.keys())
print("Student dictionary keys:", student_keys)

# Get the dictionary values as a list
student_values = list(student.values())
print("Student dictionary values:", student_values)


dog dictionary: {'name': 'preethi', 'color': 'white', 'breed': 'Labrador', 'legs': 4, 'age': 3}
student dictionary: {'first_name': 'sai', 'last_name': 'yeminedi', 'gender': 'Male', 'age': 24, 'marital_status': 'Single', 'skills': ['Python', 'Java'], 'country': 'USA', 'city': 'kansas', 'address': '525 E armour blvd'}
Length of student dictionary: 9
Skills: ['Python', 'Java']
Data type of skills: <class 'list'>
updated skills: ['Python', 'Java', 'JavaScript', 'HTML']
Student dictionary keys: ['first_name', 'last_name', 'gender', 'age', 'marital_status', 'skills', 'country', 'city', 'address']
Student dictionary values: ['sai', 'yeminedi', 'Male', 24, 'Single', ['Python', 'Java', 'JavaScript', 'HTML'], 'USA', 'kansas', '525 E armour blvd']


Create a tuple containing names of your sisters and your brothers (imaginary siblings are
fine)
• Join brothers and sisters tuples and assign it to siblings
• How many siblings do you have?
• Modify the siblings tuple and add the name of your father and mother and assign it to
family_members


In [15]:
# Create a tuple containing names of your sisters and your brothers
sisters = ('vyshu', 'puri')
brothers = ('sai', 'manoj')

# Join brothers and sisters tuples and assign it to siblings
siblings = sisters + brothers
print("siblings:", siblings)

# How many siblings do you have?
num_siblings = len(siblings)
print("Number of siblings:", num_siblings)

# Modify the siblings tuple and add the name of your father and mother and assign it to family_members
father = "chenna kesavulu"
mother = "naga jyothi"
family_members = siblings + (father, mother)

# Print the family members
print("Family members:", family_members)


siblings: ('vyshu', 'puri', 'sai', 'manoj')
Number of siblings: 4
Family members: ('vyshu', 'puri', 'sai', 'manoj', 'chenna kesavulu', 'naga jyothi')


it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}
age = [22, 19, 24, 25, 26, 24, 25, 24]
• Find the length of the set it_companies
• Add 'Twitter' to it_companies
• Insert multiple IT companies at once to the set it_companies
• Remove one of the companies from the set it_companies
• What is the difference between remove and discard
• Join A and B
• Find A intersection B
• Is A subset of B
• Are A and B disjoint sets
• Join A with B and B with A
• What is the symmetric difference between A and B
• Delete the sets completely
• Convert the ages to a set and compare the length of the list and the set.

In [20]:
# Given sets and list
it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}
age = [22, 19, 24, 25, 26, 24, 25, 24]

# Find the length of the set it_companies
len_it_companies = len(it_companies)
print("Length of it_companies:", len_it_companies)

# Add 'Twitter' to it_companies
it_companies.add('Twitter')
print("it_companies after adding 'Twitter':", it_companies)

# Insert multiple IT companies at once to the set it_companies
it_companies.update(['Netflix', 'Adobe', 'Salesforce'])
print("it_companies after adding multiple companies:", it_companies)

# Remove one of the companies from the set it_companies
it_companies.remove('Oracle')
print("it_companies after removing 'Oracle':", it_companies)

# Difference between remove and discard
# `remove` will raise an error if the element is not found
# `discard` will not raise an error if the element is not found
# Example:
# it_companies.remove('NonExistentCompany')  # This would raise a KeyError
it_companies.discard('NonExistentCompany')  # This does nothing

# Join A and B
A_union_B = A.union(B)
print("A union B:", A_union_B)

# Find A intersection B
A_intersection_B = A.intersection(B)
print("A intersection B:", A_intersection_B)

# Is A subset of B
A_subset_B = A.issubset(B)
print("Is A subset of B:", A_subset_B)

# Are A and B disjoint sets
A_disjoint_B = A.isdisjoint(B)
print("Are A and B disjoint sets:", A_disjoint_B)

# Join A with B and B with A (which is essentially the same as union)
A_join_B = A.union(B)
B_join_A = B.union(A)
print("A joined with B:", A_join_B)
print("B joined with A:", B_join_A)

# What is the symmetric difference between A and B
A_symmetric_difference_B = A.symmetric_difference(B)
print("Symmetric difference between A and B:", A_symmetric_difference_B)

# Delete the sets completely
del A
del B
print("A and B sets deleted.")

# Convert the ages to a set and compare the length of the list and the set
age_set = set(age)
print("Length of age list:", len(age))
print("Length of age set:", len(age_set))


Length of it_companies: 7
it_companies after adding 'Twitter': {'Microsoft', 'Oracle', 'Google', 'Apple', 'Facebook', 'IBM', 'Amazon', 'Twitter'}
it_companies after adding multiple companies: {'Microsoft', 'Amazon', 'Google', 'Apple', 'Netflix', 'Facebook', 'IBM', 'Salesforce', 'Adobe', 'Twitter', 'Oracle'}
it_companies after removing 'Oracle': {'Microsoft', 'Amazon', 'Google', 'Apple', 'Netflix', 'Facebook', 'IBM', 'Salesforce', 'Adobe', 'Twitter'}
A union B: {19, 20, 22, 24, 25, 26, 27, 28}
A intersection B: {19, 20, 22, 24, 25, 26}
Is A subset of B: True
Are A and B disjoint sets: False
A joined with B: {19, 20, 22, 24, 25, 26, 27, 28}
B joined with A: {19, 20, 22, 24, 25, 26, 27, 28}
Symmetric difference between A and B: {27, 28}
A and B sets deleted.
Length of age list: 8
Length of age set: 5


The radius of a circle is 30 meters.
• Calculate the area of a circle and assign the value to a variable name of _area_of_circle_
• Calculate the circumference of a circle and assign the value to a variable name of
_circum_of_circle_
• Take radius as user input and calculate the area.

In [21]:
import math

# Given radius
radius = 30

# Calculate the area of the circle
_area_of_circle_ = math.pi * radius ** 2
print("Area of the circle with radius 30 meters:", _area_of_circle_)

# Calculate the circumference of the circle
_circum_of_circle_ = 2 * math.pi * radius
print("Circumference of the circle with radius 30 meters:", _circum_of_circle_)

# Take radius as user input and calculate the area
user_radius = float(input("Enter the radius of the circle: "))
user_area_of_circle = math.pi * user_radius ** 2
print(f"Area of the circle with radius {user_radius} meters:", user_area_of_circle)


Area of the circle with radius 30 meters: 2827.4333882308138
Circumference of the circle with radius 30 meters: 188.49555921538757
Enter the radius of the circle: 10
Area of the circle with radius 10.0 meters: 314.1592653589793


“I am a teacher and I love to inspire and teach people”
• How many unique words have been used in the sentence? Use the split methods and set
to get the unique words.

In [9]:
# Given sentence
sentence = "I am a teacher and I love to inspire and teach people"

# Split the sentence into words
words = sentence.split()

# Convert the list of words to a set to get unique words
unique_words = set(words)

# Find the number of unique words
num_unique_words = len(unique_words)

# Print the number of unique words
print("Number of unique words:", num_unique_words)


Number of unique words: 10


Use a tab escape sequence to get the following lines.
Name Age Country City
Asabeneh 250 Finland Helsinki

In [10]:
# Using tab escape sequence to format the lines
print("Name\t\tAge\tCountry\t\tCity")
print("Asabeneh\t250\tFinland\t\tHelsinki")


Name		Age	Country		City
Asabeneh	250	Finland		Helsinki


Use the string formatting method to display the following:
radius = 10
area = 3.14 * radius ** 2
“The area of a circle with radius 10 is 314 meters square.”


In [11]:
# Given radius
radius = 10

# Calculate the area of the circle
area = 3.14 * radius ** 2

# Use string formatting to create the desired output
output = "The area of a circle with radius {} is {} meters square.".format(radius, area)

# Print the result
print(output)


The area of a circle with radius 10 is 314.0 meters square.


Write a program, which reads weights (lbs.) of N students into a list and convert these weights to
kilograms in a separate list using Loop. N: No of students (Read input from user)
Ex: L1: [150, 155, 145, 148]
Output: [68.03, 70.3, 65.77, 67.13]

In [24]:
# Read the number of students from the user
N = int(input("Enter the number of students: "))

# Initialize an empty list to store the weights in pounds
weights_lbs = []

# Read the weights of the students
for i in range(N):
    weight = float(input(f"Enter the weight of student {i+1} in lbs: "))
    weights_lbs.append(weight)

# Convert the weights to kilograms
weights_kg = []
for weight_lbs in weights_lbs:
    weight_kg = weight_lbs * 0.45359237
    weights_kg.append(round(weight_kg, 2))



# Print the results
print("Weights in kilograms:", weights_kg)


Enter the number of students: 4
Enter the weight of student 1 in lbs: 150
Enter the weight of student 2 in lbs: 155
Enter the weight of student 3 in lbs: 145
Enter the weight of student 4 in lbs: 148
Weights in kilograms: [68.04, 70.31, 65.77, 67.13]
